cmd/ipfs-gateway-fs/main.go

```diff
@@ -53,13 +65,19 @@ func run() error {
 			return err
 		}
 	}
-
 	repo := fsrepo.At(repoPath)
 	if err := repo.Open(); err != nil { // owned by node
 		return err
 	}
-
-	node, err := core.NewIPFSNode(ctx, core.Online(repo))
+	s3, err := makeS3Datastore()
+	if err != nil {
+		return err
+	}
+	enhanced, err := enhanceDatastore(s3)
+	if err != nil {
+		return err
+	}
+	node, err := core.NewIPFSNode(ctx, core.OnlineWithRouting(repo, corerouting.GrandCentralServer(enhanced)))
```

Comment:
does this mean all gateways will use `S3` to back up everything? it would be nice to be able to control this: only pin certain things we specifically want. Otherwise anyone could abuse this by having our gateways download TB of stuff.

How it should work:
- ipfs gateways are a cache and GC
- a subset (or a different set) pin some objects we chose to seed indefinitely